src/UsgsAstroFrameSensorModel.cpp

```diff
@@ -840,7 +817,13 @@ void UsgsAstroFrameSensorModel::replaceModelState(const std::string& stringState
         m_sensorName = state.at("m_sensorName").get<std::string>();
         m_collectionIdentifier = state.at("m_collectionIdentifier").get<std::string>();
         // Set reference point to the center of the image
-        m_referencePointXyz = imageToGround(csm::ImageCoord(m_nLines / 2.0, m_nSamples / 2.0));
+        m_referencePointXyz = imageToGround(csm::ImageCoord(m_nLines/2.0, m_nSamples/2.0));
+        // Compute the pixel ground size at the reference point
+        csm::EcefCoord nextRefPoint = imageToGround(csm::ImageCoord(m_nLines/2.0 + 1, m_nSamples/2.0 + 1));
+        double dx = nextRefPoint.x - m_referencePointXyz.x;
+        double dy = nextRefPoint.y - m_referencePointXyz.y;
+        double dz = nextRefPoint.z - m_referencePointXyz.z;
+        m_pixelGroundSize = sqrt((dx*dx + dy*dy + dz*dz) / 2.0);
```

Comment:
It is worth noting that this is not going to be accurate for highly oblique images and most flyby/disk resolved images. We can compute it more precisely for a specific groundpoint/imagepoint but I am mirroring the implementation from the LS here.

Reply:
@twilson271828 fixed this in ISIS and has a good write up on an [issue](https://github.com/USGS-Astrogeology/ISIS3/issues/2205) there

Reply:
It is worth noting. We have been asked by a few groups and scientists here to implement a more accurate resolution. That is exactly why the new one was done.

Reply:
It sounds like it may be better to calculate it at the specific pixel location instead of once for the whole image then.

I'll work on a generic pixel resolution method